File: src/main/java/de/cplaiz/activecraftcore/commands/InvseeCommand.java
package de.cplaiz.activecraftcore.commands;

import de.cplaiz.activecraftcore.ActiveCraftPlugin;
import de.cplaiz.activecraftcore.exceptions.ActiveCraftException;
import de.cplaiz.activecraftcore.utils.ComparisonType;
import org.bukkit.Sound;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.List;

public class InvseeCommand extends ActiveCraftCommand {

    public InvseeCommand(ActiveCraftPlugin plugin) {
        super("invsee",  plugin);
    }

    @Override
    public void runCommand(CommandSender sender, Command command, String label, String[] args) throws ActiveCraftException {
        checkArgsLength(args, ComparisonType.GREATER_EQUAL, 1);
        checkPermission(sender);
        Player player = getPlayer(sender);
        Player target = getPlayer(args[0]);
        player.openInventory(target.getInventory());
        player.playSound(player.getLocation(), Sound.BLOCK_CHEST_OPEN, 1f, 1f);
        messageFormatter.setTarget(getProfile(target));
        sendMessage(sender, this.cmdMsg("invsee"));
    }

    @Override
    public List<String> onTab(CommandSender sender, Command command, String label, String[] args) {
        return args.length == 1 ? getBukkitPlayernames() : null;
    }
}

File: src/main/java/org/activecraft/activecraftcore/commands/InvseeCommand.kt
package org.activecraft.activecraftcore.commands

import org.activecraft.activecraftcore.ActiveCraftPlugin
import org.activecraft.activecraftcore.exceptions.ActiveCraftException
import org.activecraft.activecraftcore.utils.ComparisonType
import org.bukkit.Sound
import org.bukkit.command.Command
import org.bukkit.command.CommandSender

class InvseeCommand(plugin: ActiveCraftPlugin) : ActiveCraftCommand("invsee", plugin) {
    @Throws(ActiveCraftException::class)
    public override fun runCommand(sender: CommandSender, command: Command, label: String, args: Array<String>) {
        assertArgsLength(args, ComparisonType.GREATER_EQUAL, 1)
        assertCommandPermission(sender)
        val player = getPlayer(sender)
        val target = getPlayer(args[0])
        player.openInventory(target.inventory)
        player.playSound(player.location, Sound.BLOCK_CHEST_OPEN, 1f, 1f)
        messageFormatter.setTarget(getProfile(target))
        sendMessage(sender, this.cmdMsg("invsee"))
    }

    public override fun onTab(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ) = if (args.size == 1) getBukkitPlayernames() else null

}